eclair-core/src/main/scala/fr/acinq/eclair/channel/Helpers.scala

```diff
@@ -443,8 +427,12 @@ object Helpers {
 
     def firstClosingFee(commitments: Commitments, localScriptPubkey: ByteVector, remoteScriptPubkey: ByteVector, feeEstimator: FeeEstimator, feeTargets: FeeTargets)(implicit log: LoggingAdapter): Satoshi = {
       val requestedFeerate = feeEstimator.getFeeratePerKw(feeTargets.mutualCloseBlockTarget)
-      // we "MUST set fee_satoshis less than or equal to the base fee of the final commitment transaction"
-      val feeratePerKw = requestedFeerate.min(commitments.localCommit.spec.feeratePerKw)
+      val feeratePerKw = if (commitments.channelVersion.hasAnchorOutputs) {
+        requestedFeerate
+      } else {
+        // we "MUST set fee_satoshis less than or equal to the base fee of the final commitment transaction"
+        requestedFeerate.min(commitments.localCommit.spec.feeratePerKw)
+      }
```

Comment:
I can't see where in the spec there is a special case for anchor outputs? It only says:

> MUST set fee_satoshis less than or equal to the base fee of the final commitment transaction, as calculated in BOLT #3.

Reply:
It's really an implementation choice, not a spec thing. This is because we enforce a low feerate on the commit tx (decorrelated from on-chain fees) so applying that `min` doesn't make any sense.

If we force the commit tx feerate to be 10 sat/byte but the on-chain feerate for our `mutualCloseBlockTarget` is higher, we should use that higher value instead of using 10 sat/byte, otherwise the mutual close will never confirm in time. Is that clearer? I'll add a comment in the code.

Reply:
I get your point, but maybe the spec needs to be updated then, because technically we are not compliant here?

Reply:
Maybe, the spec could be amended with a `SHOULD` instead of a `MUST`, I'll open a PR.

Reply:
Done in https://github.com/lightningnetwork/lightning-rfc/pull/847